service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -262,6 +271,36 @@ public WsmIamRole getHighestRole(UUID uuid, AuthenticatedUserRequest userRequest
     return highestRole.get();
   }
 
+  // If requester only has discoverer role, they can only see a subset of workspace. For example,
+  // they can see workspace name but not description.
+  private Workspace stripWorkspaceForRequesterWithOnlyDiscovererRole(Workspace fullWorkspace) {
+    Workspace.Builder strippedWorkspace =
+        new Workspace.Builder()
+            .workspaceId(fullWorkspace.getWorkspaceId())
+            .userFacingId(fullWorkspace.getUserFacingId())
+            .workspaceStage(fullWorkspace.getWorkspaceStage());
+    if (fullWorkspace.getDisplayName().isPresent()) {
+      strippedWorkspace.displayName(fullWorkspace.getDisplayName().get());
+    }
+    if (fullWorkspace.getSpendProfileId().isPresent()) {
+      strippedWorkspace.spendProfileId(fullWorkspace.getSpendProfileId().get());
```

Comment:
this seems like a strange field to make available?

Reply:
Done